### DISCUSSION TITLE
There should be some output options for the results

### DISCUSSION THREAD

**Discussion by SvenRtbg** [composer-unused/composer-unused#178] [Ideas]
As much as I like this tool, simple plain text output does not always cut it.

My task is to extract just the unused packages to add them to a problem report, for humans to read. But I might also be tempted to experimentally remove unused packages and see if this breaks anything (like counter-verify by using composer-require-checker or running tests that used to be green).

The current plain text output is rather hard to parse with shell tools, so I'd like to see some alternative output options:
- JSON (or XML if unavoidable) to either have the three categories with it's lists available, or the other way around, stating for each package the category it is in.
- Filtering of categories. I probably am not that interested in packages that are used.

Comparing this to composer, when I run `composer show`, I can select to only see `--outdated` or `--direct` dependencies.

**Comment by icanhazstring** [comment]
That sounds like an interesting idea 👍
For the moment I would go for the simple solution to output each dependency with its category as you mentioned.

Other output parsers could be added in the future.

**Comment by bradsi** [comment]
I've just started using this package to help rewrite a legacy project and clean up tech debt, starting with removing unused dependencies.

My plan is to build a command that will run this package (along with others) on a weekly basis and act as a health check, sending a report via email or integrated into an IDP.

To achieve this, I'd like to receive the output in JSON format.

@icanhazstring I searched the issues for this and noticed that you authored PR #279 to add the output format option. Could this code be extended to produce JSON output? I'm happy to contribute where I can, but I'm quite junior so would really appreciate your thoughts on this.

**Reply by icanhazstring** [reply]
Hi @bradsi.

Absolutely. You would have to create a `JsonFormatter` alongside the `GithubFormatter`.  

So you can easily add any formatting you want.
If you create a PR i am more than Happy to Support you with it und merge it 👍

**Reply by SvenRtbg** [reply]
No promises, but seeing that I asked for something like this quite some time ago, and also seeing the option included now, it might happen.

Let's discuss what the JSON output should look like. My proposal at the top suggested something like this:

    {
        "unused": [
            "package/a"
        ],
        "ignored": [],
        "zombie": []
    }

**Reply by icanhazstring** [reply]
Looks good to me 👍
